### DISCUSSION TITLE
How to use currentTiddler inside the filter to the ListWidget or list-links macro?

### DISCUSSION THREAD

**Discussion by rybak** [TiddlyWiki/TiddlyWiki5#8002] [Q&A] [answered]
_The alternative question/title to this discussion is "How to emulate MediaWiki's categorization system in TiddlyWiki5?"_

The tag system in TiddlyWiki5 is very nice. I like how every tag can have a corresponding tiddler, which crucially can be tagged itself. In fact, I've (ab)used this functionality to produce something that resembles [MediaWiki's category system](https://www.mediawiki.org/wiki/Help:Categories). For example, I created a tiddler titled `Category:Journal` with the following TiddlyWiki 5 wikitext:

```
<<list-links filter:"[tag[Category:Journal]]">>
```
which creates a viewable listing of all tiddlers tagged with `Category:Journal`, which _doesn't require clicking on the tag pill_. The listing will include other tags, whose tiddlers are tagged with `Category:Journal`; while the tiddler `Category:Journal` can then be tagged with `Category:Texts`, for example. See also [documentation of `<<list-links>>` macro](https://tiddlywiki.com/#list-links%20Macro).

This works in principle, but this approach requires retyping the name of the tag every time. It sounds like something which should be solvable with [variable `currentTiddler`](https://tiddlywiki.com/#currentTiddler%20Variable) or [`{{!!title}}`](https://tiddlywiki.com/#Current%20Tiddler) (both are similar to [MediaWiki's `{{FULLPAGENAME}}`](https://www.mediawiki.org/wiki/Help:Magic_words#Page_names)), but I couldn't figure it out, none of the following work:

```
<<list-links filter:[tag[{{!!title}}]]>>
<<list-links filter:"[tag[{{!!title}}]]">>
<<list-links filter:[tag[<<currentTiddler>>]]>>
<<list-links filter:"[tag[<<currentTiddler>>]]">>
```

I tried using [procedures](https://tiddlywiki.com/#Procedure%20Definitions) for it, but nothing I tried made it work:

```tiddlywiki
\procedure myCategoryView(name:"Parameter name is not defined")

* Plain text in procedure works: <<currentTiddler>>, <<name>>
* Can even generate a tag pill: <<tag $(currentTiddler)$ >>, but not with procedure's parameters: <<tag <<name>> >>

List-links doesn't work in either variant:

1 – `<<currentTiddler>>`
<<list-links filter:"[tag[<<currentTiddler>>]]">>

2 – `$(currentTiddler)$`
<<list-links filter:"[tag[$(currentTiddler)$]]">>

3 – `{{!!title}}`
<<list-links filter:"[tag[{{!!title}}]]">>

4 – with procedure's parameters `<<name>>`
<<list-links filter:"[tag[<<name>>]]">>


\end

----
!!! Using the procedure:

<<myCategoryView name:"<<storyTiddler>>">>

----
<<myCategoryView name:"{{!!title}}">>
```

**Comment by rybak** [comment]
I've also tried using the [ListWidget](https://tiddlywiki.com/#ListWidget), without success:

```
\procedure myCategoryView(name:"Name is not defined")

1 – `<<currentTiddler>>`
<$list filter="[tag[<<currentTiddler>>]sort[title]]"/>

2 – `$(currentTiddler)$`
<$list filter="[tag[$(currentTiddler)$]sort[title]]"/>

3 – `{{!!title}}`
<$list filter="[tag[{{!!title}}]sort[title]]"/>

4 – with procedure's parameters `<<name>>`
<$list filter="[tag[<<name>>]sort[title]]"/>

\end

----
!!! Using the procedure:

<<myCategoryView name:"<<storyTiddler>>">>

----
<<myCategoryView name:"{{!!title}}">>
```

**Comment by rybak** [comment, marked as answer]
Figured out were I went wrong: while inside a filter, the variables have a different syntax, with single angle brackets and without square brackets. See section "Special parameters" in [Introduction to filter notation](https://tiddlywiki.com/#Introduction%20to%20filter%20notation).

The fixed wikitext is then:

```diff
<<list-links filter:"[tag<currentTiddler>]">>

<$list filter="[tag<currentTiddler>sort[title]]"/>
```

Though output of ListWidget needs heavy tweaking to resemble output of `<<list-links>>`:

```
<ul>
  <$list filter="[tag<currentTiddler>sort[title]]">
    <li><$link to=<<currentTiddler>> /></li>
  </$list>
</ul>
```

**Comment by Jermolene** [comment]
Hi @rybak glad you got it figured out. Questions about wikitext may get more response at https://talk.tiddlywiki.org

**Reply by rybak** [reply]
> Questions about wikitext may get more response at https://talk.tiddlywiki.org/

Thanks, I've posted [a follow-up](https://talk.tiddlywiki.org/t/filter-to-list-all-tiddlers-which-have-a-word-in-their-title-but-dont-have-the-word-or-another-word-in-their-tags/9197) there.
